README.md
# Docklight
Docklight is a lightweight Dock launcher for common Applications.

Docklight” is a simple but effective dock launcher for GNOME desktops e.g (gnome, cinnamon, mate etc.).
It supports a Preview from all open Applications.

The “Docklight” preview allows more control over the opened windows. You can navigate to a specific window much easier.
Select the window or close the desired application the real time preview make that very easy.
“Docklight” is written in C++ with “GTK 3” and “libwnck”. “Docklight” is open source width a GNU General Public License.


![docklight_sample](https://cloud.githubusercontent.com/assets/9448387/19643317/e3d066ea-9a13-11e6-8971-96cf38752c97.jpg)


Docklight has been designed for users who have to handle multiple applications. The desktop may have several windows open or hidden behind other windows. This is where docklight can help you handle multiple windows easily.


 Licensing:

  This program is released under the terms of the GNU General Public
  License (GNU GPL).

  You can find a copy of the license in the file LICENSE.
  
 Compiling:
  
  In order to build this program you need a working GNOME environment
  version 3 or greater, with the development tools installed properly.
  The following is the detailed list of libraries you need:
  
	- gtkmm-3.0
	- libwnck-3.0	
	- libx11-dev
	- Autotols
   
	./autogen.sh
	./configure --prefix=/usr/lib/docklight
	make
	sudo make install


 Install from a debian package:
 
  you can install “Docklight” direct from the debian Package/docklight_1.0-*_amd64.deb
 
 Credits:

  Docklight is based of the inspiration from other great docks. It is simple but is still in   development. 
  It will  bring more refinements and  new features to make Docklight more comfortable ad powerful.
  Many thanks to the all developers and to all the people who contribute to debian.

  -eof-






debian/changelog
docklight (1.0-19) UNRELEASED; urgency=medium

  * add selector easing animation
  * several bug fixing	 

 -- yoo <yoo@osx.odx.com>  Fri, 10 Mar 2017 15:21:04 +0700

docklight (1.0-18) UNRELEASED; urgency=medium

  * several bug fixes 

 -- yoo <yoo@osx.odx.com>  Sun, 05 Mar 2017 00:22:03 +0700

docklight (1.0-17) UNRELEASED; urgency=medium

  * Add images folder
  * fixes in attach to dock 
  * Modify debian install script
  * Modidy autotool data/Makefile.am
 	 
 -- yoo <yoo@osx.odx.com>  Thu, 02 Mar 2017 18:52:28 +0700

docklight (1.0-16) UNRELEASED; urgency=medium

  * Add Session Group and Persistence
  * Several bug fixes
  * Modify po Translations files	 

 -- yoo <yoo@osx.odx.com>  Sat, 25 Feb 2017 17:42:23 +0700

docklight (1.0-15) UNRELEASED; urgency=medium

  * Buf Fix in selector background alpha channel 

 -- yoo <yoo@osx.odx.com>  Mon, 20 Feb 2017 19:49:56 +0700

docklight (1.0-14) UNRELEASED; urgency=medium

  * add Languages es,de,th
  * add Themes 

 -- yoo <yoo@osx.odx.com>  Wed, 15 Feb 2017 19:08:13 +0700

docklight (1.0-13) UNRELEASED; urgency=medium

  * create drag & drop functionality
  * fix minor bug fixes	

 -- Juan González <j-gonzalez@email.de>  Tue, 31 Jan 2017 17:24:50 +0700

docklight (1.0-12) UNRELEASED; urgency=medium

  * create autotools build system
  * prepare for translations 

 -- Juan R. González <j-gonzalez@email.de>  Sun, 22 Jan 2017 22:36:29 +0700

docklight (1.0-11) UNRELEASED; urgency=medium

  * Modify the Preview.
  * Add dynamic and static image loading.
  * Several bug fixings in preview.
  * Move preview update to Position namespace. 

 -- Juan R. González <j-gonzalez@email.de>  Sun, 15 Jan 2017 20:32:00 +0700

docklight (1.0-10) UNRELEASED; urgency=medium

  * reorganize launcher and utilities
  * add Home Menu items for minimize and unminimized
  * add Items unminimized Menu 	 

 -- Juan R. González <j-gonzalez@email.de>  Tue, 10 Jan 2017 18:36:41 +0700

docklight (1.0-9) UNRELEASED; urgency=medium

  * test: debian package installation
  * test: on linux mint
  * test: on ubuntu 

 -- Juan R. González <j-gonzalez@email.de>  Sun, 08 Jan 2017 02:44:20 +0700

docklight (1.0-8) UNRELEASED; urgency=medium

  * debian file an ITP (Intent To Package) 

 -- Juan R. González <j-gonzalez@email.de>  Fri, 06 Jan 2017 14:54:05 +0700

docklight (1.0-7) UNRELEASED; urgency=medium

  * add package comments and remove litian errors and warnings
	
 -- Juan R. González <j-gonzalez@email.de>  Fri, 06 Jan 2017 14:52:52 +0700

docklight (1.0-6) UNRELEASED; urgency=medium

  * test: debian package purge 

 -- Juan R. González <j-gonzalez@email.de>  Fri, 06 Jan 2017 14:52:21 +0700

docklight (1.0-5) UNRELEASED; urgency=medium

  * test: debian package creation

 -- Juan R. González <j-gonzalez@email.de>  Fri, 06 Jan 2017 14:51:54 +0700

docklight (1.0-4) UNRELEASED; urgency=medium

  * modify make file 

 -- Juan R. González <j-gonzalez@email.de>  Fri, 06 Jan 2017 14:51:22 +0700

docklight (1.0-3) UNRELEASED; urgency=medium

  * add debian deb scripts 

 -- Juan R. González <j-gonzalez@email.de>  Fri, 06 Jan 2017 14:50:50 +0700

docklight (1.0-2) UNRELEASED; urgency=medium

  * organize install process 

 -- Juan R. González <j-gonzalez@email.de>  Fri, 06 Jan 2017 14:50:06 +0700

docklight (1.0.1) UNRELEASED; urgency=medium

  * Initial release: prepare for debian 

 -- Juan R. González <j-gonzalez@email.de>  Fri, 06 Jan 2017 14:49:40 +0700


src/Defines.h
//*****************************************************************
//
//  Copyright (C) 2015 Juan R. Gonzalez
//  Created on November 20, 2015, 12:17 PM 
//
//  This program is free software: you can redistribute it and/or modify
//  it under the terms of the GNU General Public License as published by
//  the Free Software Foundation, either version 3 of the License, or
//  (at your option) any later version.
// 
//  This program is distributed in the hope that it will be useful,
//  but WITHOUT ANY WARRANTY; without even the implied warranty of
//  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
//  GNU General Public License for more details.
// 
//  You should have received a copy of the GNU General Public License
//  along with this program.  If not, see <http://www.gnu.org/licenses/>.
//
//****************************************************************

#ifndef DEFINES_H
#define DEFINES_H

#define DEF_DATADIRNAME         "data/" 
#define DEF_ATTACHMENTDIR       "data/attachments" 
#define DEF_IMAGESDIR           "data/images" 
#define DEF_ICONNAME            "data/images/docklight.home.ico"
#define DEF_SEISSIONICONNAME    "data/images/docklight-session.png"
#define DEF_LOGONAME            "data/images/docklight.logo.png"
#define DEF_INITNAME            "data/docklight.ini"
#define DEF_LAUCHERTEMPLATE     "docklight.template.desktop"
#define DEF_LAUCHERTEMPLATEPATH "data/docklight.template.desktop"
#define DEF_LAUCHERSCRIPT       "docklight.launcher.sh"
#define DEF_LAUCHERSCRIPTPATH   "data/docklight.launcher.sh"
#define DEF_MAXSESIONIMAGES     12
#define DEF_LOCALEPATH          "/usr/share/locale"
#define DEF_LOCALEPACKPATH      "/usr/share/locale-langpack"
#define DEF_LAUNCHER_DIR        "/usr/share/applications/"
#define DOCKLIGHT_INSTANCENAME  "docklight"
#define DOCKLIGHTLAUNCHER       "DockLight Launcher"

#define NULLPB (Glib::RefPtr<Gdk::Pixbuf>)NULL

typedef enum Window_action_t
{
    OPEN,
    CLOSE
} Window_action;

typedef enum panel_location_t
{
    TOP,
    BOTTOM
} panel_locationType;

typedef enum StrutsPosition_t
{
    Left = 0,
    Right = 1,
    Top = 2,
    Bottom = 3,
    LeftStart = 4,
    LeftEnd = 5,
    RightStart = 6,
    RightEnd = 7,
    TopStart = 8,
    TopEnd = 9,
    BottomStart = 10,
    BottomEnd = 11
} strutsPosition;




#define DEF_FRAMERATE 1000/30

#define DEF_ICONMAXSIZE 64
#define DEF_CELLHIGHT  56                    // Cell Height
#define DEF_CELLWIDTH DEF_CELLHIGHT -3//58                    // Cell Width
#define DEF_CELLTOPMARGIN 6                 // Cell Top Position
#define DEF_ICONSIZE DEF_CELLHIGHT -14      // Icon Size Width and height
#define DEF_MINCONSIZE 20                   // Minimun Icon Size allowed
#define DEF_ICONTOPMARGIN 10                // Icon top Margin from panel

// Icon top Margin from the cell
#define DEF_ICONTOPCELLMARGIN DEF_ICONTOPMARGIN -DEF_CELLTOPMARGIN                

#define DEF_PANELHIGHT DEF_CELLHIGHT + 9
#define DEF_PANELHIGHTHIDE  4                 



#define DEF_PREVIEW_WIDTH 220               // the preview popup window width
#define DEF_PREVIEW_HEIGHT 200              // the preview popup window height
#define DEF_PREVIEW_MINHEIGHT 112           // the minimun preview height
#define DEF_PREVIEW_LEFT_MARGING 14         // the preview left marging 
#define DEF_PREVIEW_RIGHT_MARGING 30        // the preview right marging 
#define DEF_PREVIEW_SCALE_WIDTH_OFFSET  12  // the offset for the scale method
#define DEF_PREVIEW_SCALE_HEIGHT_OFFSET 54  // the offset for the scale method
#define DEF_PREVIEW_PIXBUF_TOP 36           // the image pixbuf top


#endif /* DEFINES_H */

